### ISSUE TITLE
after running a git pull I get a broken prompt

### ISSUE THREAD

**Issue by ezpuzz** [nickcox/pure-pwsh#1]
Hi!  thanks for the great work! I love pure and glad to see a version for powershell.

I have an issue at the moment, when I run a `git checkout <branch>` or `git pull` that works I get a prompt with a `?` instead of the `❯`.

It also replicates the last command there, then immediately draws a new prompt.  Then my next command fails because it tries to combine it with the reprinted command.

Here is example output:

```powershell
E:\repos\virtuoso develop ⇣
❯ git pull
Updating 76e0a5f..f18bf0e
Fast-forward
 app/pods/application/template.hbs                  |   9 +-
 .../application-current-user/template.hbs          |  10 +-
 app/pods/components/application-nav/template.hbs   |  34 +++--
 .../components/application-search/template.hbs     |  18 +--
 .../components/display-flash-messages/template.hbs |   4 +-
 app/templates/head.hbs                             |   2 +-
 package.json                                       |   3 +-

E:\repos\virtuoso develop
? git pull

E:\repos\virtuoso develop  (3.05s)
❯ git pull
fatal: Cannot rebase onto multiple branches.

E:\repos\virtuoso develop  (2.51s)
❯ git status
On branch develop
Your branch is up to date with 'origin/develop'.

nothing to commit, working tree clean

E:\repos\virtuoso develop
❯ git pull
Already up to date.
Current branch develop is up to date.
```

My profile is this:
```powershell
if ($host.Name -eq 'ConsoleHost')
{
    Import-Module PSReadLine
}

Remove-PSReadlineKeyHandler 'CTRL+R'
Import-Module PSFzf

Import-Module -Name E:\repos\pure-pwsh\pure-pwsh\pure-pwsh.psd1
Import-Module posh-git
#Import-Module oh-my-posh
#Set-Theme agnoster

new-item -path Env:. -name FZF_DEFAULT_COMMAND -value "rg --files . 2> nul"
new-item -path Env:. -name EDITOR -value "atom"
New-Alias vim nvim
New-Alias which get-command

Import-Module "C:\ProgramData\chocolatey\lib\ripgrep\tools\_rg.ps1"

#function Get-GitStatus { & git status $args }
#New-Alias -Name gst -Value Get-GitStatus -Force -Option AllScope
#function Get-GitCommit { & git commit -ev $args }
#New-Alias -Name gc -Value Get-GitCommit -Force -Option AllScope
function Get-GitAdd { & git add --all $args }
New-Alias -Name ga -Value Get-GitAdd -Force -Option AllScope
function Get-GitTree { & git lg }
New-Alias -Name lg -Value Get-GitTree -Force -Option AllScope
function Get-GitCheckout { & git checkout $args }
New-Alias -Name gco -Value Get-GitCheckout -Force -Option AllScope
function Get-GitBranch { & git branch $args }
New-Alias -Name gb -Value Get-GitBranch -Force -Option AllScope
function Get-GitDiff { & git diff $args }
New-Alias -Name gd -Value Get-GitDiff -Force -Option AllScope
# Chocolatey profile
$ChocolateyProfile = "$env:ChocolateyInstall\helpers\chocolateyProfile.psm1"
if (Test-Path($ChocolateyProfile)) {
  Import-Module "$ChocolateyProfile"
}

$DefaultUser = 'invad'

# Host Foreground
$Host.PrivateData.ErrorForegroundColor = 'Red'
$Host.PrivateData.WarningForegroundColor = 'Yellow'
$Host.PrivateData.DebugForegroundColor = 'Green'
$Host.PrivateData.VerboseForegroundColor = 'Blue'
$Host.PrivateData.ProgressForegroundColor = 'Gray'

# Host Background
$Host.PrivateData.ErrorBackgroundColor = 'DarkGray'
$Host.PrivateData.WarningBackgroundColor = 'DarkGray'
$Host.PrivateData.DebugBackgroundColor = 'DarkGray'
$Host.PrivateData.VerboseBackgroundColor = 'DarkGray'
```

**Comment by nickcox** [comment]
Thanks for the feedback. I probably should add a note to the readme to make it clear that I'm still ironing some issues.

I think this happens because it's trying to re-render the prompt before it's done writing the new one. Should be able to get a fix in over the next few days.

**Comment by nickcox** [comment]
I've just published ~0.1.3~ [0.1.4](https://www.powershellgallery.com/packages/pure-pwsh/0.1.4). It's working pretty well for me now but I'd very much appreciate any further feedback.

**Comment by ezpuzz** [comment]
working great for me now, will report if i see any other weird behavior
